src/info.c

```diff
@@ -0,0 +1,179 @@
+/****************************************************************************
+**
+**  This file is part of GAP, a system for computational discrete algebra.
+**
+**  Copyright of GAP belongs to its developers, whose names are too numerous
+**  to list here. Please refer to the COPYRIGHT file for details.
+**
+**  SPDX-License-Identifier: GPL-2.0-or-later
+**
+**  This file declares the functions handling Info statements.
+*/
+
+
+#include "info.h"
+
+#include "bool.h"
+#include "calls.h"
+#include "error.h"
+#include "gapstate.h"
+#include "gvars.h"
+#include "modules.h"
+#include "plist.h"
+
+#ifdef HPCGAP
+#include "hpc/aobjects.h"
+#endif
+
+enum {
+    INFODATA_NUM = 1,
+    INFODATA_CURRENTLEVEL,
+    INFODATA_CLASSNAME,
+    INFODATA_HANDLER,
+    INFODATA_OUTPUT,
+};
+
+static Obj InfoDecision;
+static Obj IsInfoClassListRep;
+static Obj DefaultInfoHandler;
+static Obj ResetShowUsedInfoClassesHandler;
+static Obj ShowUsedInfoClassesHandler;
+
+
+Obj FuncShowUsedInfoClasses(Obj self, Obj choice)
+{
+    if (choice == True) {
+        STATE(ShowUsedInfoClassesActive) = 1;
+        CALL_0ARGS(ResetShowUsedInfoClassesHandler);
+    }
+    else if (choice == False) {
+        STATE(ShowUsedInfoClassesActive) = 0;
+    }
+    else {
+        ErrorMayQuit("<setting> should be true or false", 0, 0);
```

Comment:
Could also use `RequireTrueOrFalse` for a consistent error message

Reply:
... and then a test of this error handling would also be nice, for full coverage.